### PR TITLE
Add TC feature definitions to General Commissioning Cluster XML files

### PR DESCRIPTION
**Pull request by swan-amazon**
This commit introduces the "Terms and Conditions" (TC) feature definitions to the General Commissioning Cluster XML data model files. The changes include new attributes, commands, and commissioning errors related to the TC feature.

Changes include:
- Added TC feature definition with bitmask.
- Introduced new commissioning errors related to TC:
  - RequiredTCNotAccepted
  - TCAcknowledgementsNotReceived
  - TCMinVersionNotMet
- Added new attributes for TC:
  - TCAcceptedVersion
  - TCMinRequiredVersion
  - TCAcknowledgements
  - TCAcknowledgementsRequired
- Added new commands for TC acknowledgements:
  - SetTCAcknowledgements
  - SetTCAcknowledgementsResponse

These changes only modify the XML files. A follow-up commit will regenerate the necessary files.

### CODE REVIEW

**Comment by swan-amazon** [comment]
Question: How do I resolve these warnings that are thrown when I try to zap regen?

```
WARNING: Missing "introduced" or "provisional" entry for: 'GeneralCommissioning' '{"deprecationMessage":"The errorCode field will be removed","commandField":"errorCode","command":"SetTCAcknowledgementsResponse"}'
WARNING: Missing "introduced" or "provisional" entry for: 'GeneralCommissioning' '{"deprecationMessage":"The TCAcceptedVersion attribute will be removed","isForIds":true,"minimalRelease":"267F4B03-3256-4056-A62D-5237640FDCFE","attribute":"TCAcceptedVersion"}'
WARNING: Missing "introduced" or "provisional" entry for: 'GeneralCommissioning' '{"deprecationMessage":"The TCMinRequiredVersion attribute will be removed","isForIds":true,"minimalRelease":"267F4B03-3256-4056-A62D-5237640FDCFE","attribute":"TCMinRequiredVersion"}'
WARNING: Missing "introduced" or "provisional" entry for: 'GeneralCommissioning' '{"deprecationMessage":"The TCAcknowledgements attribute will be removed","isForIds":true,"minimalRelease":"267F4B03-3256-4056-A62D-5237640FDCFE","attribute":"TCAcknowledgements"}'
WARNING: Missing "introduced" or "provisional" entry for: 'GeneralCommissioning' '{"deprecationMessage":"The TCAcknowledgementsRequired attribute will be removed","isForIds":true,"minimalRelease":"267F4B03-3256-4056-A62D-5237640FDCFE","attribute":"TCAcknowledgementsRequired"}'
WARNING: Missing "introduced" or "provisional" entry for: 'GeneralCommissioning' '{"attribute":"TCAcceptedVersion"}'
WARNING: Missing "introduced" or "provisional" entry for: 'GeneralCommissioning' '{"attribute":"TCMinRequiredVersion"}'
WARNING: Missing "introduced" or "provisional" entry for: 'GeneralCommissioning' '{"attribute":"TCAcknowledgements"}'
WARNING: Missing "introduced" or "provisional" entry for: 'GeneralCommissioning' '{"attribute":"TCAcknowledgementsRequired"}'
WARNING: Missing "introduced" or "provisional" entry for: 'GeneralCommissioning' '{"minimalRelease":"267F4B03-3256-4056-A62D-5237640FDCFE","attribute":"TCAcceptedVersion"}'
WARNING: Missing "introduced" or "provisional" entry for: 'GeneralCommissioning' '{"minimalRelease":"267F4B03-3256-4056-A62D-5237640FDCFE","attribute":"TCAcceptedVersion"}'
WARNING: Missing "introduced" or "provisional" entry for: 'GeneralCommissioning' '{"minimalRelease":"267F4B03-3256-4056-A62D-5237640FDCFE","attribute":"TCAcceptedVersion"}'
WARNING: Missing "introduced" or "provisional" entry for: 'GeneralCommissioning' '{"minimalRelease":"267F4B03-3256-4056-A62D-5237640FDCFE","attribute":"TCMinRequiredVersion"}'
WARNING: Missing "introduced" or "provisional" entry for: 'GeneralCommissioning' '{"minimalRelease":"267F4B03-3256-4056-A62D-5237640FDCFE","attribute":"TCMinRequiredVersion"}'
WARNING: Missing "introduced" or "provisional" entry for: 'GeneralCommissioning' '{"minimalRelease":"267F4B03-3256-4056-A62D-5237640FDCFE","attribute":"TCMinRequiredVersion"}'
WARNING: Missing "introduced" or "provisional" entry for: 'GeneralCommissioning' '{"minimalRelease":"267F4B03-3256-4056-A62D-5237640FDCFE","attribute":"TCAcknowledgements"}'
WARNING: Missing "introduced" or "provisional" entry for: 'GeneralCommissioning' '{"minimalRelease":"267F4B03-3256-4056-A62D-5237640FDCFE","attribute":"TCAcknowledgements"}'
WARNING: Missing "introduced" or "provisional" entry for: 'GeneralCommissioning' '{"minimalRelease":"267F4B03-3256-4056-A62D-5237640FDCFE","attribute":"TCAcknowledgements"}'
WARNING: Missing "introduced" or "provisional" entry for: 'GeneralCommissioning' '{"minimalRelease":"267F4B03-3256-4056-A62D-5237640FDCFE","attribute":"TCAcknowledgementsRequired"}'
WARNING: Missing "introduced" or "provisional" entry for: 'GeneralCommissioning' '{"minimalRelease":"267F4B03-3256-4056-A62D-5237640FDCFE","attribute":"TCAcknowledgementsRequired"}'
WARNING: Missing "introduced" or "provisional" entry for: 'GeneralCommissioning' '{"minimalRelease":"267F4B03-3256-4056-A62D-5237640FDCFE","attribute":"TCAcknowledgementsRequired"}'
WARNING: Missing "introduced" or "provisional" entry for: 'GeneralCommissioning' '{"deprecationMessage":"The tcVersion field will be removed","commandField":"tcVersion","command":"SetTCAcknowledgements"}'
WARNING: Missing "introduced" or "provisional" entry for: 'GeneralCommissioning' '{"deprecationMessage":"The tcUserResponse field will be removed","commandField":"tcUserResponse","command":"SetTCAcknowledgements"}'
```

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
@swan-amazon Note that this will fail CI until codegen is run.

File: data_model/master/clusters/GeneralCommissioningCluster.xml [hunk 1]
```diff
@@ -65,7 +65,7 @@ Davis, CA 95616, USA
   </clusterIds>
   <classification hierarchy="base" role="utility" picsCode="CGEN" scope="Node"/>
   <features>
-    <feature bit="0" code="TC" name="Terms" summary="Supports Terms &amp; Conditions acknowledgement">
+    <feature bit="0" code="TC" name="Terms and Conditions" summary="Supports Terms and Conditions acknowledgement">
```
Comment (cecille) [inline data_model/master/clusters/GeneralCommissioningCluster.xml:68]: data_model files are scraped from the spec and should not be manually updated.  please remove these changes from the PR.

I should also check why the CI checker isn't flagging these...

Comment (swan-amazon) [reply data_model/master/clusters/GeneralCommissioningCluster.xml]: I reverted the changes to this file.

File: src/app/zap-templates/zcl/data-model/chip/general-commissioning-cluster.xml [hunk 2]
```diff
@@ -83,5 +109,15 @@ limitations under the License.
       <arg name="ErrorCode" type="CommissioningErrorEnum"/>
       <arg name="DebugText" type="char_string"/>
     </command>
+    <command source="client" code="0x06" name="SetTCAcknowledgements" optional="true" response="SetTCAcknowledgementsResponse" cli="chip fabric_commissioning settcacknowledgements" apiMaturity="provisional">
```
Comment (tcarmelveilleux) [inline src/app/zap-templates/zcl/data-model/chip/general-commissioning-cluster.xml:112]: The `cli` stuff is not needed and completely obsolete... You can leave it for now but please don't cut and paste things you are not sure about...

**Review by tcarmelveilleux — APPROVED** [review]

**Pull request merged**